src/lib/yang/data.lua

```diff
@@ -1245,6 +1510,142 @@ function rpc_output_printer_from_schema(schema)
    return data_printer_from_grammar(rpc_output_grammar_from_schema(schema))
 end
 
+local function influxdb_printer_tests ()
+   local function lint (text)
+      local ret = {}
+      for line in text:gmatch("[^\n]+") do
+         table.insert(ret, (line:gsub("^%s+", "")))
+      end
+      return table.concat(ret, "\n")
+   end
+   local function influxdb_printer_test (test)
+      local schema_str, data_str, expected = unpack(test)
+      local format = 'influxdb'
+      local is_config, print_default = true, true
+      local schema = schema.load_schema(schema_str)
+      local data = load_config_for_schema(schema, mem.open_input_string(data_str))
+      local grammar = data_grammar_from_schema(schema, is_config, format)
+      local printer = influxdb_printer_from_grammar(grammar, print_default)
+      local actual = mem.call_with_output_string(printer, data)
+      assert(actual == lint(expected))
+   end
+   local test_schema = [[
+      module test {
+         namespace test;
+         prefix test;
+
+         container foo {
+            leaf x { type string; }
+            leaf y { type string; }
+            list bar {
+               key baz;
+               leaf baz { type string; }
+               leaf y { type string; }
+               leaf z { type string; }
+            }
+         }
+         container continents {
+            grouping country {
+               list country {
+                  key name;
+                  leaf name { type string; mandatory true; }
+                  leaf capital { type string; }
+                  leaf gdp { type decimal64; }
+                  leaf eu-member { type boolean; }
+                  leaf main-cities { type string; }
+               }
+            }
+            container europe {
+               uses country;
+            }
+            container asia {
+               uses country;
+            }
+         }
+         container users {
+            leaf-list allow-user {
+               type string;
+            }
+         }
+         container nested-list {
+            leaf-list foo {
+               type string;
+            }
+            list bar {
+               leaf-list foo {
+                  type string;
+               }
+            }
+         }
+      }
+   ]]
+
+   local tests = {
+      {test_schema,
+      [[
+         foo {
+            x "x";
+            y "y";
+            bar {baz "baz"; y "y"; z "z";}
+         }
+      ]], [[
+         foo/bar/y,baz=baz value="y"
```

Comment:
For the uniquified names, WDYT about adding an initial / prefix?  That would be a nice visual indicator that we're dealing with an absolute path.